docs/haskell-use-cases.rst

```diff
@@ -3,6 +3,64 @@
 Common Haskell Build Use Cases
 ==============================
 
+Picking a compiler
+------------------
+
+Unlike Bazel's native C++ rules, rules_haskell does not auto-detect
+a Haskell compiler toolchain from the environment. This is by design.
+We require that you declare a compiler to use in your ``WORKSPACE``
+file.
+
+There are two common sources for a compiler. One is to use the
+official binary distributions from `haskell.org`_. This is done using
+the `ghc_bindist`_ rule.
+
+The compiler can also be pulled from Nixpkgs_, a set of package
+definitions for the `Nix package manager`_. Pulling the compiler from
+Nixpkgs makes the build more hermetic, because the transitive closure
+of the compiler and all its dependencies is precisely defined in the
+``WORKSPACE`` file. Use `rules_nixpkgs`_ to do so (where ``X.Y.Z``
+stands for any recent release)::
+
+  http_archive(
```

Comment:
We could avoid the repetition of the release name with something like

```python
nixpkgs_release = "X.Y.Z"
http_archive(
    name = "io_tweag_rules_nixpkgs",
    strip_prefix = "rules_nixpkgs-{}".format(nixpkgs_release),
    urls = ["https://github.com/tweag/rules_nixpkgs/archive/v{}.tar.gz".format(nixpkgs_release)],
)

Reply:
For the example it’s easier to see with the dummy, it acts like a “meta”-variable already, non?

Reply:
@Profpatsch agreed.